Remiworks/Remiworks.Core/Event/EventListener.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Remiworks.Core.Event.Matching;
+using Remiworks.Core.Models;
+
+namespace Remiworks.Core.Event
+{
+    public class EventListener : IEventListener
+    {
+        private readonly object _lockObject = new object();
+
+        private readonly Dictionary<string, List<CallbackForTopic>> _queueCallbacks;
+        private readonly IBusProvider _busProvider;
+
+        public EventListener(IBusProvider busProvider)
+        {
+            _queueCallbacks = new Dictionary<string, List<CallbackForTopic>>();
+            _busProvider = busProvider;
+        }
+
+        public async Task SetupQueueListenerAsync<TParam>(string queueName, EventReceived<TParam> callback)
+        {
+            await SetupQueueListenerAsync(
```

Comment:
De task returnen, zodat het nut heeft om deze functie te awaiten. Deze is public dus dan kan die weer door een andere aangeroepen worden die het miss async wilt laten uitvoeren of wilt awaiten

Reply:
Hier zit ik nog wel een beetje over te twijfelen, maar miss iets om over na te denken